### PR TITLE
[NEW] App `onInstall` event

### PR DESCRIPTION
**Pull request by lucassartor**
# What? :boat:

**Added:** `onInstall` event method.

# Why? :thinking:

# Links :earth_americas:

[ClickUp Task](https://app.clickup.com/t/98bbxj)
Fixes  https://github.com/RocketChat/Rocket.Chat.Apps-engine/issues/397

# PS :eyes:
Complements https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/407

### CODE REVIEW

File: src/server/AppManager.ts [hunk 1]
```diff
@@ -733,6 +735,31 @@ export class AppManager {
...
+
+        try {
+            await app.call(AppMethod.ONINSTALL, read, http, persistence, user);
+
+            result = true;
+        } catch (e) {
+            const status = AppStatus.ERROR_DISABLED;
+
+            if (e.name === 'NotEnoughMethodArgumentsError') {
+                console.warn('Please report the following error:');
```
Comment (lolimay) [inline src/server/AppManager.ts:752]: Don't simply use `console.warn`, use `app.getLogger().warn` instead. Please append the original error to the log as well.

Comment (lucassartor) [reply src/server/AppManager.ts]: Ok I've fixed it. I used `console.warn` because other methods in this file use it, my bad!

**Review by d-gubert — CHANGES_REQUESTED** [review]

File: src/server/AppManager.ts [hunk 2]
```diff
@@ -414,7 +414,7 @@ export class AppManager {
         return true;
     }
 
-    public async add(appPackage: Buffer, installationParameters: IAppInstallParameters): Promise<AppFabricationFulfillment> {
+    public async add(appPackage: Buffer, installationParameters: IAppInstallParameters, user: IUser): Promise<AppFabricationFulfillment> {
```
[the remark this answers is not in the document]
Comment (lucassartor) [reply src/server/AppManager.ts]: It really does. I've fixed it, thanks!

File: src/server/AppManager.ts [hunk 3]
```diff
@@ -733,6 +735,31 @@ export class AppManager {
         return this.enableApp(storageItem, app, true, isManual, silenceStatus);
     }
 
+    private async installApp(storageItem: IAppStorageItem, app: ProxiedApp, user: IUser): Promise<boolean> {
+        let result: boolean;
```
Comment (thassiov) [inline src/server/AppManager.ts:740]: Maybe lose this variable and return a boolean directly in the `try/catch` block? We won't run any code after it, after all. Does it make sense?

**Review by thassiov — COMMENTED** [review]
LGTM. There's just the context thing left to put in the method's argument, right?
**EDIT**: nvm, for some odd reason I saw only 3 files instead of 5, and the context were in the two that disappeared :thinking:

**Review by d-gubert — APPROVED** [review]

File: src/server/AppManager.ts (hunk 3 above)
Comment (d-gubert) [reply src/server/AppManager.ts:740]: Let's review this during the beta period

**Pull request merged**